clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/tracing/RedisDynamicSamplingStrategy.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2018 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.clouddriver.aws.tracing;
+
+import com.amazonaws.xray.strategy.sampling.SamplingRule;
+import com.amazonaws.xray.strategy.sampling.SamplingRuleManifest;
+import com.amazonaws.xray.strategy.sampling.SamplingStrategy;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.PropertyNamingStrategy;
+import com.netflix.spinnaker.kork.jedis.RedisClientSelector;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import rx.Scheduler;
+import rx.schedulers.Schedulers;
+
+import javax.annotation.PostConstruct;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.ThreadLocalRandom;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Allows loading dynamic sampling rules from Redis.
+ *
+ * This is largely lifted from LocalizedSamplingStrategy, all the way down to
+ * the Yoda conditionals, which only allows rules from a file location. :(
+ *
+ * If any rules are defined in Redis, it is treated like a full rule
+ * definition. The fallback manifest will only be used if no key in Redis
+ * exists.
+ */
+public class RedisDynamicSamplingStrategy implements SamplingStrategy {
+
+  private final static Logger log = LoggerFactory.getLogger(RedisDynamicSamplingStrategy.class);
+
+  private final static String CLIENT_NAME = "tracing";
+  private final static String KEY = "{tracing:xray}:samplingRules";
+
+  private final RedisClientSelector redisClientSelector;
+  private final SamplingRuleManifest fallbackManifest;
```

Comment:
Rule manifests look like this: https://github.com/aws-samples/eb-java-scorekeep/blob/xray/src/main/resources/sampling-rules.json